apps/opik-backend/src/main/java/com/comet/opik/api/resources/v1/priv/TracesResource.java

```
@@ -689,4 +693,58 @@ public Response closeTraceThread(
+    @Path("/threads/feedback-scores")
+    @Operation(operationId = "scoreBatchOfThreads", summary = "Batch feedback scoring for threads", description = "Batch feedback scoring for threads", responses = {
+            @ApiResponse(responseCode = "204", description = "No Content")})
+    @RateLimited
+    public Response scoreBatchOfThreads(
+            @RequestBody(content = @Content(schema = @Schema(implementation = FeedbackScoreBatch.class))) @JsonView(FeedbackScoreBatch.View.Thread.class) @Validated(FeedbackScoreBatch.View.Thread.class) @NotNull @Valid FeedbackScoreBatch batch) {
+
+        String workspaceId = requestContext.get().getWorkspaceId();
+
+        log.info("Feedback scores batch for threads, size '{}' on  workspaceId '{}'", batch.scores().size(),
+                workspaceId);
+
+        feedbackScoreService.scoreBatchOfThreads(batch.scores())
+                .contextWrite(ctx -> setRequestContext(ctx, requestContext))
+                .retryWhen(AsyncUtils.handleConnectionError())
+                .block();
+
+        log.info("Feedback scores batch for threads, size '{}' on  workspaceId '{}'", batch.scores().size(),
+                workspaceId);
+
+        return Response.noContent().build();
+    }
+
+    @POST
+    @Path("/threads/feedback-scores/delete")
+    @Operation(operationId = "deleteThreadFeedbackScores", summary = "Delete thread feedback scores", description = "Delete thread feedback scores", responses = {
+            @ApiResponse(responseCode = "204", description = "No Content")})
+    public Response deleteThreadFeedbackScores(
+            @RequestBody(content = @Content(schema = @Schema(implementation = DeleteThreadFeedbackScores.class))) @NotNull @Valid DeleteThreadFeedbackScores scores) {
+        var workspaceId = requestContext.get().getWorkspaceId();
+        String projectName = scores.projectName();
+
+        Optional<Project> project = projectService.findByNames(workspaceId, List.of(projectName)).stream().findFirst();
```

we have to find a way to hide the logic in the service (here we have another point where logic been done on the resources level) - any suggestions?

---

We have cases where we use a facade, for instance, the experiment build API, which handles many domain services. We need to be careful when adding services as dependencies of other services. As a rule, it's not critical, but this frequently leads to cycle dependencies. 

In this case, I will change, but before implementing such a practice, I would open a discussion with the team. In cases like this, it doesn't add much value to introducing facades. Usually, `facade` coordinates complex operations across different domain operations, simplifies the APIs, and reduces the complexity of dealing with such operations. In this case, we are referring to a simple 'fail fast' exit using an already existing dependency of the controller. Again, no big deal changing it, but it's been suggested before starting to implement it in all system.